File: Example/PDStrategy/PDStrategy/PDStrategy/Models/PDRule.m
//
//  PDRule.m
//  PDStrategy
//
//  Created by Pavel Deminov on 19/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

#import "PDRule.h"

@implementation PDRule

- (instancetype)init {
    self = [super init];
    if (self) {
        [self setup];
    }
    return self;
}

- (void)setup {
    
}

- (BOOL)validate:(id)value {
    if (self.validationBlock) {
        return self.validationBlock(value);
    } else {
        return true;
    }
}

@end

File: Example/PDStrategy/PDStrategy/PDStrategy/Models/PDRule.swift
//
//  PDRule.swift
//  PDStrategy
//
//  Created by Pavel Deminov on 04/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

import UIKit


typealias ValidationBlock = (_ value: Any) -> Bool

class PDRule: NSObject {
    var validationBlock: ValidationBlock?
    var error = ""
    
    func validate(_ value: Any) -> Bool {
        if let validationBlock = validationBlock {
            return validationBlock(value)
        }
        else {
            return true
        }
    }
}

